### PR TITLE
Resourcer Middleware

### PR DESCRIPTION
**Pull request by sirlori**
The README explains a lot.
If you have any question I will happily answer asap!

Thanks and bye!

### CODE REVIEW

**Comment by sirlori** [comment]
Also [here](https://github.com/sirlori/gin) I have my gin's fork.
 You can see that there's only a little change in the Context struct. 
I know that conceptually this is not so good, but we can try to solve this issue together I think.
I mean have to make accessible the Resource in the handler and I don't see how if not this.
But I will be really chuffed to see some hints on this.

**Comment by manucorporat** [comment]
c.Set("gin.contrib.resourcer", value)
c.Get("gin.contrib.resourcer")

We do not need to add a new variable

File: resourcer/resourcer.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		if err != nil {
+			if gin.IsDebugging() {
+				c.AbortWithError(404, err)
+			} else {
+				c.AbortWithStatus(404)
+			}
+		}
+		c.Resource = res
+		// Before request
+		c.Next()
```
Comment (manucorporat) [inline resourcer/resourcer.go]: c.Next() is not needed.

File: resourcer/resourcer.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+func ResourcerGorm(h GormHandlers, db *gorm.DB) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		// Gets resource and sets it into Context
+		res, err := h.Resource(h, db, c.Params)
+
+		if err != nil {
+			if gin.IsDebugging() {
+				c.AbortWithError(404, err)
```
Comment (manucorporat) [inline resourcer/resourcer.go]: Always `c.AbortWithError(404, err)` should be used.

Comment (manucorporat) [inline resourcer/resourcer.go]: Also, c.Abort() does not jump to the end of the scope. `return;` should be added.

**Comment by sirlori** [comment]
OK I have fixed the mess that I had pushed.
Sorry but I only started yesterday to work with this, although I'm really enjoying gin, Great Work!

**Pull request closed without merging**